### ISSUE TITLE
Error on clicking Exit Game

### ISSUE THREAD

**Issue by Farmbuyer** [erglo/wow-mission-report-button-plus#30]
This occurs when logging out in a new TWW zone while not in a rested area -- i.e., the 20-second countdown appears instead of instantly leaving the game.

```
1x ...rfaceMissionReportButtonPlus/data/labels.lua:316: attempt to index global 'MRBP_GlobalSettings' (a nil value)
[string "@MissionReportButtonPlus/data/labels.lua"]:316: in function `CleanUpLabels'
[string "@MissionReportButtonPlus/MissionReportButtonPlus.lua"]:120: in function <.../MissionReportButtonPlus/MissionReportButtonPlus.lua:107>

Locals:
self = <table> {
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index global 'MRBP_GlobalSettings' (a nil value)"
L = <table> {
 CFG_DDMENU_NAMING_TEXT = "Prefer Expansion Names"
 CFG_ADDONINFOS_AUTHOR = "Author"
 showMawThreats = "Covenant Assaults"
 CFG_DDMENU_ENTRYTOOLTIP_GARRISON_INVASION_ALERT_TOOLTIP = "Shows an alert when an invasion in your Garrison is active."
 showLegionAssaultsInfo = "Legion Assault"
 CFG_DDMENU_ENTRYTOOLTIP_LABEL = "Details Tooltip"
 CFG_DDMENU_ENTRYTOOLTIP_MISSION_INFO_TOOLTIP = "Displays a brief summary of your currently active missions you started at the mission table."
 CFG_MAJOR_FACTION_SEPARATE_TOOLTIP_TEXT = "Use Separate Tooltip"
 CFG_DDMENU_NAMING_TOOLTIP = "The dropdown menu items are by default the names of each expansion.|n|nIf disabled, the name of each mission report will be displayed instead."
 ENTRYTOOLTIP_TIMEWALKING_VENDOR_LABEL = "Timewalking Vendor"
 ENTRYTOOLTIP_WORLD_MAP_EVENTS_LABEL = "World Map Events"
 CFG_DDMENU_ENTRYTOOLTIP_MAW_THREATS_TOOLTIP = "Displays a brief summary on the Covenant Assaults in The Maw."
 CFG_DDMENU_ENTRYTOOLTIP_SHOW_TOOLTIP = "On mouse-over each menu item details about the corresponding expansion are shown in a tooltip.|n|nIf disabled, this tooltip will not appear."
 applyInvasionColors = "Apply Invasion Colors"
 CFG_APPEARANCE_ANCHOR_DISTANCE_TOOLTIP = "Set the spacing between the menu frame and the minimap button."
 CFG_DDMENU_ENTRYTOOLTIP_FACTION_COLORS_TOOLTIP = "Displays the faction names in their respective colors."
 showGrandHuntsInfo = "Grand Hunts"
 CFG_DDMENU_STYLESELECTION_LABEL = "Menu Style"
 CFG_DDMENU_ENTRYTOOLTIP_DRAGON_GLYPHS_TOOLTIP = "Shows the progress of your Dragon Glyphs collection of each zone."
 CFG_DDMENU_STYLESELECTION_VALUE1_TEXT = "Tooltip Style"
 CFG_DDMENU_ENTRYTOOLTIP_BFA_BOUNTIES_TOOLTIP = "Lists the currently active Emissary Quests available on Zandalar and Kul Tiras in the tooltip."
 showCovenantBounties = "Callings"
 applyBfAFactionColors = "Apply Faction Colors"
 ENTRYTOOLTIP_LEGION_APPLY_INVASION_COLORS_LABEL = "Apply Invasion Colors"
 CFG_DDMENU_ENTRYTOOLTIP_EVENT_POI_ISKAARA_FEAST = "the Community Feast of the Iskaara Tuskarr"
 TOC_FILE_NOTES = "Adds a right-click menu to the mission reports button on the minimap|nwith access to reports from current and previous expansions."
 showBfAFactionAssaultsInfo = "Faction Assaults"
 CFG_DDMENU_ENTRYTOOLTIP_EVENT_POI_TIME_RIFTS = "the Time Rifts in Thaldraszus"
 CFG_DDMENU_ENTRYSELECTION_TEXT_D = "%d |4menu item:menu items; selected"
 showNzothThreats = "N'Zoth Assaults"
 CFG_DDMENU_ENTRYTOOLTIP_EVENT_POI_HIDE_EVENT_DESCRIPTIONS = "Hides additional event description texts when enabled."
 CFG_DDMENU_ENTRYTOOLTIP_EVENT_POI_SUPERBLOOM = "the Superbloom in the Emerald Dream"
 CFG_DDMENU_ENTRYTOOLTIP_EVENT_POI_DREAMSURGE = "the Dreamsurge on the Dragon Isles"
 CFG_APPEARANCE_DIMENSIONS_LABEL = "Dimensions"
 CFG_APPEARANCE_ANCHOR_POINT_MENU_TOOLTIP = "Set the menu frame's anchor point."
 CFG_APPEARANCE_ANCHOR_LABEL = "Anchor Point"
 CFG_DDMENU_SORTORDER_TOOLTIP = "The dropdown menu items are sorted by expansion release date; eg. the current expansion comes first, the one before that next, etc.|n|nIf enabled, the sorting order will be reversed."
 CFG_APPEARANCE_DIMENSIONS_FRAME_WIDTH_TOOLTIP = "Set the minimum width for the menu frame."
 autoHideCompletedDragonGlyphZones = "Hide Completed Zones"
 CFG_MINIMAPBUTTON_SHOWBUTTON_TOOLTIP = "As long as you haven't reached the highest level available for an extension by WoW default the minimap button for the expansion/garrison landing page remains hidden.|n|nIf enabled, this option shows you the button again with access to your last command table along with additional informations."
 showWoDWorldMapEvents = "World Map Events"
 showLegionWorldMapEvents = "World Map Events"
 CFG_DDMENU_ENTRYTOOLTIP_LEGION_INVASION_COLORS_TOOLTIP = "Display the invasion names in a light green color."
 CFG_SHOW_ADDON_COMPARTMENT_TEXT = "Addon Compartment"
 showWoDGarrisonInvasionAlert = "Garrison Invasion"
 CF
```

**Comment by erglo** [comment]
Hi, thank you for reporting this!
I assume that this occurred with version `v1.2.1` and not the pre-release version from the `development` branch?! I'm going to release most probably tomorrow a new version of this addon. Since I'm not getting any errors currently I assume this error you encountered just occurred due to the lack of proper data needed for `The War Within`.
I'll test this behavior, just in case.

**Comment by Farmbuyer** [comment]
Affirmative, v1.2.1 in the .toc file, retrieved via the CurseForge app (so, however they do their packaging etc).

**Comment by erglo** [comment]
Hi, I fixed this.
I added some clean-up routines to keep the variables files "tidy" when logging out or exiting the game, but when you try to use the minimap button's menu during this process or after aborting said process, the addon tried to access tables which had already been removed during clean-up. Now the routine re-initializes what it needs which seems to solve the problem.
- Thanks again for pointing this out!
